pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -248,62 +249,54 @@ func (c *Cluster) removeExtraMgrs(daemonIDs []string) {
-		return "", err
-	} else if prevActiveMgr == currActiveMgr {
-		logger.Infof("active mgr is still the same (%s). No need to update mgr_role label on daemon %s.", currActiveMgr, daemonNameToUpdate)
-		return currActiveMgr, err
-	}
-
+// SetMgrRoleLabel sets 'mgr_role: active' label to given manager daemon pods if isActive is true.
+// Otherwise sets 'mgr_role: standby' label to manager pods.
+func (c *Cluster) SetMgrRoleLabel(daemonNameToUpdate string, isActive bool) error {
 	pods, err := c.context.Clientset.CoreV1().Pods(c.clusterInfo.Namespace).List(c.clusterInfo.Context, metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("mgr=%s", daemonNameToUpdate),
+		LabelSelector: fmt.Sprintf("%s=%s", controller.DaemonIDLabel, daemonNameToUpdate),
 	})
 	if err != nil {
 		logger.Infof("cannot get pod for mgr daemon %s", daemonNameToUpdate)
-		return "", err // force mrg_role update in the next call
+		return err // force mrg_role update in the next call
 	}
 
+	newMgrRole := standbyMgrStatus
+	if isActive {
+		newMgrRole = activeMgrStatus
+	}
 	// Normally, there should only be one mgr pod with the specific name daemonNameToUpdate. However,
 	// during transitions, there might be additional mgr pods shutting down. To handle this, the code
 	// updates the label mgrRoleLabelName on all mgr pods. If this update fails, the system rolls back
 	// the currently active manager (currActiveMgr). This way the next call will retry the update.
+	var podLabelUpdErr error
 	for i, pod := range pods.Items {
-
 		labels := pod.GetLabels()
-		cephDaemonId := labels[controller.DaemonIDLabel]
-		newMgrRole := standbyMgrStatus
-		if currActiveMgr == cephDaemonId {
-			newMgrRole = activeMgrStatus
-		}
-
 		currMgrRole, mgrHasLabel := labels[mgrRoleLabelName]
 		if !mgrHasLabel || currMgrRole != newMgrRole {
-
-			logger.Infof("updating mgr_role label value of daemon %s to '%s'. New active mgr is %s.", daemonNameToUpdate, newMgrRole, currActiveMgr)
+			logger.Infof("updating mgr_role label value of daemon %s to '%s'. New active mgr is %s.", daemonNameToUpdate, newMgrRole, daemonNameToUpdate)
 			labels[mgrRoleLabelName] = newMgrRole
 			pod.SetLabels(labels)
 			_, err = c.context.Clientset.CoreV1().Pods(c.clusterInfo.Namespace).Update(c.clusterInfo.Context, &pods.Items[i], metav1.UpdateOptions{})
 			if err != nil {
-				// In case of failure we report as 'active manager' the previous value, this way
-				// we force refreshing mgrRoleLabelName label next time this function is called
-				currActiveMgr = prevActiveMgr
-				logger.Warningf("cannot update the active mgr pod %q. err=%v", pods.Items[i].Name, err)
+				// don't return error from here. First try to update all pods from the list and reconcile services.
+				// return error later to update pods on next reconcile.
+				podLabelUpdErr = srderrors.Join(podLabelUpdErr, fmt.Errorf("cannot update the active mgr pod %q. err=%w", pods.Items[i].Name, err))
```

Comment:
Maybe `errors.Wrapf()` will be fine, no need for another errors package
```suggestion
				podLabelUpdErr = errors.Wrapf(podLabelUpdErr, fmt.Errorf("cannot update the active mgr pod %q. err=%w", pods.Items[i].Name, err))
```

Reply:
Or to keep it simple, just assign the last error since 99% of the time there is only one pod being updated anyway. Either approach is ok for me.
```suggestion
				podLabelUpdErr = fmt.Errorf("cannot update the active mgr pod %q. err=%w", pods.Items[i].Name, err)
```